src/popsift/popsift.cpp

```diff
@@ -19,9 +19,13 @@
 
 using namespace std;
 
-PopSift::PopSift( const popsift::Config& config, popsift::Config::ProcessingMode mode, ImageMode imode )
+PopSift::PopSift( const popsift::Config& config, popsift::Config::ProcessingMode mode, ImageMode imode, int device )
     : _image_mode( imode )
+    , _device(device)
 {
+    cudaSetDevice(_device);
```

Comment:
This looks like a good thing to do. Perhaps with an error check when a user chooses an non-existant device?

Reply:
I agree. Do you think using `device_prop_t::set(int, bool)` would be good? Another solution would be manual check with
```
POP_CUDA_FATAL_TEST( cudaSetDevice( currentDevice ), "Cannot set device" );
```
or maybe just `POP_CHK`.